From beeccfbaed230f21ffce83a54f258c01790bb4a7 Mon Sep 17 00:00:00 2001
From: Robin Jarry <robin@jarry.cc>
Date: Thu, 21 Mar 2024 14:57:37 +0100
Subject: [PATCH] meson: add dependency definition

Signed-off-by: Robin Jarry <robin@jarry.cc>
---
 meson.build | 5 +++++
 1 file changed, 5 insertions(+)
 create mode 100644 meson.build

diff --git a/meson.build b/meson.build
new file mode 100644
index 000000000000..4b60f15fe423
--- /dev/null
+++ b/meson.build
@@ -0,0 +1,5 @@
+project('stb', 'c', version : '0.1', license : 'MIT or Public domain')
+
+stb_dep = declare_dependency(
+  include_directories: include_directories('.'),
+)
-- 
2.44.0

